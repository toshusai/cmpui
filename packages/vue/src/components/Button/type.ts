export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
