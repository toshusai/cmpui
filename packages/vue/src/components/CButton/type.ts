export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
