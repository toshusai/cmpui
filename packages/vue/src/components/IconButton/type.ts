export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
