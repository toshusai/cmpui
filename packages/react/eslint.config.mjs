"use strict";
import eslintConfigPrettier from "eslint-config-prettier";
import tseslint from "typescript-eslint";
import eslintPluginImport from "eslint-plugin-import";

export default [
  ...tseslint.configs.recommended,
  {
    ignores: ["dist/*", "node_modules/*", "storybook-static/*"],
  },
  {
    plugins: { import: eslintPluginImport },
    rules: {
      "vue/multi-word-component-names": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": ["error"],
      "import/order": [
        "error",
        {
          "newlines-between": "always",
        },
      ],
    },
    languageOptions: {
      parserOptions: { parser: tseslint.parser, sourceType: "module" },
    },
  },
  eslintConfigPrettier,
];
