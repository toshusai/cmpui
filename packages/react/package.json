{
  "name": "@toshusai/cmpui",
  "version": "1.0.0-alpha.13",
  "description": "CmpUI is a component library designed for creating complex GUI applications in web browsers.",
  "main": "dist/index.mjs",
  "types": "dist/index.d.mts",
  "scripts": {
    "typecheck": "tsc -p tsconfig.ci.json",
    "test:watch": "vitest",
    "test": "vitest run",
    "dev": "tsup --watch",
    "build": "tsup",
    "storybook": "storybook dev -p 6007",
    "storybook:build": "storybook build",
    "storybook:test": "test-storybook",
    "storybook:test:ci": "concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"pnpm storybook:build && npx http-server storybook-static --port 6006 --silent\" \"wait-on http://127.0.0.1:6006 && pnpm storybook:test\"",
    "format": "prettier .",
    "format:fix": "prettier --write .",
    "lint": "eslint .",
    "lint:fix": "eslint --fix .",
    "fix": "npm run format:fix && npm run lint:fix"
  },
  "dependencies": {
    "@floating-ui/dom": "^1.6.10",
    "@radix-ui/react-context-menu": "^2.1.5",
    "@radix-ui/react-dropdown-menu": "^2.0.6",
    "@radix-ui/react-menubar": "^1.0.4",
    "@radix-ui/react-select": "^2.0.0",
    "@radix-ui/react-toast": "^1.1.5",
    "@toshusai/cmpui-core": "link:../core",
    "@toshusai/cmpui-css": "link:../css",
    "react-transition-group": "^4.4.5"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@playwright/test": "^1.41.2",
    "@storybook/addon-essentials": "8.2.6",
    "@storybook/addon-interactions": "8.2.6",
    "@storybook/addon-links": "8.2.6",
    "@storybook/addon-mdx-gfm": "8.2.6",
    "@storybook/addon-onboarding": "8.2.6",
    "@storybook/addon-themes": "8.2.6",
    "@storybook/blocks": "8.2.6",
    "@storybook/react": "8.2.6",
    "@storybook/react-vite": "8.2.6",
    "@storybook/test": "8.2.6",
    "@storybook/test-runner": "^0.17.0",
    "@storybook/theming": "8.2.6",
    "@tabler/icons-react": "^3.15.0",
    "@types/jest": "^29.5.11",
    "@types/react": "^18.2.37",
    "@types/react-dom": "^18.2.15",
    "@types/react-transition-group": "^4.4.11",
    "concurrently": "^8.2.2",
    "eslint": "^9.9.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-import": "^2.29.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "storybook": "8.2.6",
    "tsup": "^8.0.1",
    "typescript": "^5.2.2",
    "vitest": "^1.1.0",
    "wait-on": "^7.2.0"
  },
  "keywords": [
    "react",
    "typescript",
    "components",
    "ui",
    "gui"
  ],
  "author": "toshusai",
  "license": "MIT"
}
