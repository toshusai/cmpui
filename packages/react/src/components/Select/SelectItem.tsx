import * as RadixSelect from "@radix-ui/react-select";
import { ListItem } from "../ListItem";

export type SelectItemProps = {
  children: React.ReactNode;
  disabled?: boolean;
  onSelect?: () => void;
  value: string;
};

export function SelectItem(props: SelectItemProps) {
  return (
    <RadixSelect.Item
      value={props.value}
      asChild
      onSelect={props.onSelect}
      disabled={props.disabled}
    >
      <ListItem
        rounded
        style={{
          paddingLeft: 24,
        }}
      >
        <RadixSelect.ItemIndicator asChild>
          <svg
            style={{
              position: "absolute",
              left: 8,
            }}
            xmlns="http://www.w3.org/2000/svg"
            height="12"
            viewBox="0 -960 960 960"
            width="12"
            fill="currentColor"
          >
            <path d="M382-208 122-468l90-90 170 170 366-366 90 90-456 456Z" />
          </svg>
        </RadixSelect.ItemIndicator>
        <RadixSelect.ItemText>{props.children}</RadixSelect.ItemText>
      </ListItem>
    </RadixSelect.Item>
  );
}
