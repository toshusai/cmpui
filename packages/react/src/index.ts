export * from "./components/Button";
export * from "./components/CanvasView";
export * from "./components/Checkbox";
export * from "./components/Circle";
export * from "./components/ColorInput";
export * from "./components/ColorLoupe";
export * from "./components/ColorPickerCircle";
export * from "./components/ColorPickerRect";
export * from "./components/ColorSlider";
export * from "./components/ContextMenu";
export * from "./components/FloatBox";
export * from "./components/Grid";
export * from "./components/HueCircle";
export * from "./components/IconButton";
export * from "./components/IconButton/IconButtonGroup";
export * from "./components/Line";
export * from "./components/ListItem";
export * from "./components/MenuBar/MenuBar";
export * from "./components/MenuBar/MenuBarButton";
export * from "./components/Modal";
export * from "./components/Popover";
export * from "./components/RectGizmo";
export * from "./components/Ruler";
export * from "./components/Select";
export * from "./components/SelectRect";
export * from "./components/Slider";
export * from "./components/SliderNumberField";
export * from "./components/SplitPane";
export * from "./components/SVPicker";
export * from "./components/Switch";
export * from "./components/TextArea";
export * from "./components/TextInput";
export * from "./components/Toast";
export * from "./components/Tooltip";
export * from "./components/TreeView";
export * from "./utils";
