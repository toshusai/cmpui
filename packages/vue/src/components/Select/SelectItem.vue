<script setup lang="ts">
import MenuListItem from "../MenuListItem/MenuListItem.vue";

withDefaults(
  defineProps<{
    disabled?: boolean;
    value?: string;
  }>(),
  {
    disabled: false,
    value: undefined,
  },
);
</script>

<template>
  <MenuListItem
    :disabled="disabled"
    :data-value="value"
    role="option"
    v-bind="$attrs"
  >
    <slot></slot>
  </MenuListItem>
</template>
