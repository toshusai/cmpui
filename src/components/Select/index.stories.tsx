import type { Meta, StoryObj } from "@storybook/react";
import { useEffect, useState } from "react";
import { Select } from ".";
import { SelectItem } from "./SelectItem";

const meta: Meta<typeof Select> = {
  title: "Components/Form/Select",
  component: Select,
};

export default meta;

type Story = StoryObj<typeof Select>;

const options = [
  {
    label: "Option 1",
    value: "1",
  },
  {
    label: "Option 2",
    value: "2",
  },
  {
    label: "Option 3",
    value: "3",
  },
  {
    label: "Option 4",
    value: "4",
  },
];

export const Basic: Story = {
  render: function Render() {
    const [value, setValue] = useState("1");
    return (
      <Select label="Label" value={value} onChange={setValue}>
        {options.map((option) => (
          <SelectItem value={option.value} key={option.value}>
            {option.label}
          </SelectItem>
        ))}
      </Select>
    );
  },
};

export const Disabled: Story = {
  render: function Render() {
    const [value, setValue] = useState("1");
    return (
      <Select label="Label" value={value} onChange={setValue} disabled>
        {options.map((option) => (
          <SelectItem value={option.value} key={option.value}>
            {option.label}
          </SelectItem>
        ))}
      </Select>
    );
  },
};

export const DisabledItem: Story = {
  render: function Render() {
    const [value, setValue] = useState("1");
    return (
      <Select label="Label" value={value} onChange={setValue}>
        {options.map((option) => (
          <SelectItem
            value={option.value}
            key={option.value}
            disabled={option.value === "2"}
          >
            {option.label}
          </SelectItem>
        ))}
      </Select>
    );
  },
};

const fontList = [
  "Roboto",
  "Sixtyfour",
  "Wavefont",
  "Gabarito",
  "Agbalumo",
  "Honk",
  "Slackside One",
  "Smooch",
  "Chokokutai",
  "Monomaniac One",
  "Palette Mosaic",
].sort();
export const Fonts: Story = {
  render: function Render() {
    const fonts = fontList.map((font) => ({
      label: font,
      value: font,
      family: font,
      url: `https://fonts.googleapis.com/css2?family=${font}&display=swap`,
    }));
    const [value, setValue] = useState("Roboto");
    return (
      <Select label="font" value={value} onChange={setValue}>
        {fonts.map((font) => (
          <FontRender
            family={font.family}
            url={font.url}
            value={font.value}
            key={font.value}
          >
            {font.label}
          </FontRender>
        ))}
      </Select>
    );
  },
};

const map = new Map<string, string>();
function loadFontFromCss(url: string) {
  if (map.has(url)) return;

  const link = document.createElement("link");
  link.href = url;
  link.rel = "stylesheet";
  document.head.appendChild(link);

  map.set(url, url);
}

function FontRender(props: {
  value: string;
  url: string;
  family: string;
  children: React.ReactNode;
}) {
  useEffect(() => {
    loadFontFromCss(props.url);
  }, [props.url]);

  return (
    <SelectItem value={props.value}>
      <div
        style={{
          fontFamily: props.family,
        }}
      >
        {props.children}
      </div>
    </SelectItem>
  );
}
