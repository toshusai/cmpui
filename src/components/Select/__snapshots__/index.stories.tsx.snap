// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form/Select Basic smoke-test 1`] = `
<button type="button"
        role="combobox"
        aria-controls="radix-:r0:"
        aria-expanded="false"
        aria-autocomplete="none"
        dir="ltr"
        data-state="closed"
        class="cmpui_select__root"
>
  <span style="pointer-events: none;">
    Option 1
  </span>
  <svg xmlns="http://www.w3.org/2000/svg"
       height="16"
       viewbox="0 -960 960 960"
       width="16"
       fill="currentColor"
       aria-hidden="true"
  >
    <path d="M480-345 240-585l56-56 184 184 184-184 56 56-240 240Z">
    </path>
  </svg>
  <label class="cmpui_select__label">
    Label
  </label>
</button>
`;

exports[`Form/Select Disabled smoke-test 1`] = `
<button type="button"
        role="combobox"
        aria-controls="radix-:r5:"
        aria-expanded="false"
        aria-autocomplete="none"
        dir="ltr"
        data-state="closed"
        disabled
        data-disabled
        class="cmpui_select__root"
>
  <span style="pointer-events: none;">
    Option 1
  </span>
  <svg xmlns="http://www.w3.org/2000/svg"
       height="16"
       viewbox="0 -960 960 960"
       width="16"
       fill="currentColor"
       aria-hidden="true"
  >
    <path d="M480-345 240-585l56-56 184 184 184-184 56 56-240 240Z">
    </path>
  </svg>
  <label class="cmpui_select__label">
    Label
  </label>
</button>
`;

exports[`Form/Select DisabledItem smoke-test 1`] = `
<button type="button"
        role="combobox"
        aria-controls="radix-:ra:"
        aria-expanded="false"
        aria-autocomplete="none"
        dir="ltr"
        data-state="closed"
        class="cmpui_select__root"
>
  <span style="pointer-events: none;">
    Option 1
  </span>
  <svg xmlns="http://www.w3.org/2000/svg"
       height="16"
       viewbox="0 -960 960 960"
       width="16"
       fill="currentColor"
       aria-hidden="true"
  >
    <path d="M480-345 240-585l56-56 184 184 184-184 56 56-240 240Z">
    </path>
  </svg>
  <label class="cmpui_select__label">
    Label
  </label>
</button>
`;

exports[`Form/Select Fonts smoke-test 1`] = `
<button type="button"
        role="combobox"
        aria-controls="radix-:rf:"
        aria-expanded="false"
        aria-autocomplete="none"
        dir="ltr"
        data-state="closed"
        class="cmpui_select__root"
>
  <span style="pointer-events: none;">
    <div style="font-family: Roboto;">
      Roboto
    </div>
  </span>
  <svg xmlns="http://www.w3.org/2000/svg"
       height="16"
       viewbox="0 -960 960 960"
       width="16"
       fill="currentColor"
       aria-hidden="true"
  >
    <path d="M480-345 240-585l56-56 184 184 184-184 56 56-240 240Z">
    </path>
  </svg>
  <label class="cmpui_select__label">
    font
  </label>
</button>
`;
